### PR TITLE
chore(ci): test start-stop aws instance

### PR DESCRIPTION
**Pull request by soonum**

### CODE REVIEW

File: .github/workflows/registered_runner.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+jobs:
+  test-runner:
+    name: Test Runner
+    runs-on: ${{ inputs.runner-name }}
+    timeout-minutes: 5  # Job should be picked very quickly
+    steps:
+      - name: Checkout
+        id: checkout
+        uses: actions/checkout@eef61447b9ff4aafe5dcd4e0bbf5d482be7e7871
```
[the remark this answers is not in the document]
Comment (IceTDrinker) [reply .github/workflows/registered_runner.yml:19]: ah no ok got it

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -76,3 +81,38 @@ jobs:
...
+      - name: Test stop instance
+        id: test-stop
+        if: ${{ always() }}
+        uses: ./
+        with:
+          mode: stop
+          github-token: ${{ secrets.SLAB_ACTION_TOKEN }}
+          slab-url: ${{ secrets.SLAB_BASE_URL_PRE_PROD }}
+          job-secret: ${{ secrets.JOB_SECRET }}
+          label: ci-persistent-runner
```
Comment (IceTDrinker) [inline .github/workflows/ci.yml:118]: any way to check the runner is still there even if offline and instance did not disappear ?

**Review by IceTDrinker — APPROVED** [review]
Can be merged as is, don't know if it tests all you want though

File: .github/workflows/registered_runner.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (soonum) [reply .github/workflows/registered_runner.yml:19]: If you can run this job, this means that your runner is up and running.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (soonum) [reply .github/workflows/ci.yml:118]: I can add another sub workflow to do that.

**Comment by soonum** [comment]
> Can be merged as is, don't know if it tests all you want though

The registered_runner workflow fails, I hope this is because there is no reference of it in `main` yet.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (IceTDrinker) [reply .github/workflows/ci.yml:118]: as you wish

Comment (IceTDrinker) [reply .github/workflows/ci.yml:118]: or run the workflow twice ?

**Comment by IceTDrinker** [comment]
> > Can be merged as is, don't know if it tests all you want though
> 
> The registered_runner workflow fails, I hope this is because there is no reference of it in `main` yet.

No if it has the PR trigger it means something is really failing

**Comment by IceTDrinker** [comment]
you need to checkout before calling a workflow from a workflow

File: .github/workflows/test_spawn_terminate.yml [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+      matrix:
+        runner: [ fromJSON(needs.action-start.outputs) ]
+      fail-fast: false
+    steps:
+      - name: Checkout
+        id: checkout
+        uses: actions/checkout@eef61447b9ff4aafe5dcd4e0bbf5d482be7e7871
+
+      - name: Run job self-hosted runner
+        uses: ./.github/workflows/registered_runner.yml
```
Comment (IceTDrinker) [inline .github/workflows/test_spawn_terminate.yml:60]: this needs to be at the job level, not step level

File: .github/workflows/removed_runner.yml [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+      - name: Check runner removed
+        if: inputs.must-exist == false
+        run: |
+          ( echo "${{ env.RUNNERS_FOUND }}" | grep --line-regexp -q "${{ inputs.runner-name }}" ) && { exit 1; } || exit 0
+
+      - name: Check runner persists
+        if: inputs.must-exist == true
+        run: |
+          ( echo "${{ env.RUNNERS_FOUND }}" | grep --line-regexp -q "${{ inputs.runner-name }}" ) || exit 1
```
Comment (IceTDrinker) [inline .github/workflows/removed_runner.yml:37]: newline at the end of the file

**Pull request merged**